core/controllers/admin.py

```diff
@@ -127,57 +132,131 @@ def get(self):
                              (self.user_id, config_property_id))
                 config_services.revert_property(
                     self.user_id, config_property_id)
+                self.render_json({})
             elif self.payload.get('action') == 'start_new_job':
                 for klass in jobs_registry.ONE_OFF_JOB_MANAGERS:
                     if klass.__name__ == self.payload.get('job_type'):
                         klass.enqueue(klass.create_new())
                         break
+                self.render_json({})
             elif self.payload.get('action') == 'cancel_job':
                 job_id = self.payload.get('job_id')
                 job_type = self.payload.get('job_type')
                 for klass in jobs_registry.ONE_OFF_JOB_MANAGERS:
                     if klass.__name__ == job_type:
                         klass.cancel(job_id, self.user_id)
                         break
+                self.render_json({})
             elif self.payload.get('action') == 'start_computation':
                 computation_type = self.payload.get('computation_type')
                 for klass in jobs_registry.ALL_CONTINUOUS_COMPUTATION_MANAGERS:
                     if klass.__name__ == computation_type:
                         klass.start_computation()
                         break
+                self.render_json({})
             elif self.payload.get('action') == 'stop_computation':
                 computation_type = self.payload.get('computation_type')
                 for klass in jobs_registry.ALL_CONTINUOUS_COMPUTATION_MANAGERS:
                     if klass.__name__ == computation_type:
                         klass.stop_computation(self.user_id)
                         break
+                self.render_json({})
             elif self.payload.get('action') == 'upload_topic_similarities':
                 data = self.payload.get('data')
                 recommendations_services.update_topic_similarities(data)
+                self.render_json({})
+            elif self.payload.get('action') == 'view_by_role':
+                role = self.payload.get('role')
+                users_by_role = user_services.get_usernames_by_role(role)
+                self.render_json(users_by_role)
+            elif self.payload.get('action') == 'view_role_by_username':
+                user_name = self.payload.get('username')
+                user_id = user_services.get_user_id_from_username(user_name)
+                result = {}
+                result[user_name] = user_services.get_user_role_from_id(
+                    user_id)
+                self.render_json(result)
+            elif self.payload.get('action') == 'update_user_role':
+                user_id = user_services.get_user_id_from_username(
+                    self.payload.get('username'))
+                user_services.update_user_role(
+                    user_id, self.payload.get('role'))
+                self.render_json({})
```

Comment:
ok. Which one is a better way then -

adding,
ADMIN_ROLE_HANDLER :
    get -> handles request for view roles form.
    post -> handles request for update roles form.

or adding separate handlers (don't need to fetch roles in admin.html),
VIEW_ROLE_HANDLER:
   get -> gives viewable roles (to show in select)
   post -> handles view role query

UPDATE_ROLE_HANDLER
   get -> view updatable roles. 
   post -> handles update role query.